Spatially vary temperature in momentum initialization (#2357)
ECP-WarpX/WarpX: Docs/source/usage/parameters.rst

      ``<species_name>.uy_th`` and ``<species_name>.uz_th``. Note that the average momenta normal to the plane is not used.

    * ``maxwell_boltzmann``: Maxwell-Boltzmann distribution that takes a dimensionless

>       temperature parameter theta as an input, where theta is kb*T/(m*c^2),

Reviewer: While we are on it, let's sprinkle some LaTeX in here :)
```suggestion
      temperature parameter :math:`\theta` as an input, where :math:`\theta = \frac{k_\mathrm{B} \cdot T}{m \cdot c^2}`,
```

>       kb is the Boltzmann constant, c is the speed of light, and m is the mass of the species.

Reviewer: ```suggestion
      :math:`k_\mathrm{B}` is the Boltzmann constant, :math:`c` is the speed of light, and :math:`m` is the mass of the species.
```

      they are not relativistic in the drift frame. This is as opposed to the Maxwell Juttner
      setting, which initializes particles with relativistic momentums in their drifting frame.

>     * ``maxwell_juttner``: Maxwell-Juttner distribution for high temperature plasma that takes a dimensionless temperature parameter theta as an input, where theta is kb*T/(m*c^2), kb is the Boltzmann constant, and m is the mass of the species.

Reviewer: ```suggestion
    * ``maxwell_juttner``: Maxwell-Juttner distribution for high temperature plasma that takes a dimensionless temperature parameter :math:`\theta` as an input, where :math:`\theta = \frac{k_\mathrm{B} \cdot T}{m \cdot c^2}`, :math:`k_\mathrm{B}` is the Boltzmann constant, and :math:`m` is the mass of the species.
```

      Theta is specified by a combination of ``<species_name>.theta_distribution_type``, ``<species_name>.theta``, and ``<species_name>.theta_function(x,y,z)`` (see below).
      It also

>       includes the optional parameter ``<species_name>.beta`` where beta is equal to v/c. The plasma

Reviewer: ```suggestion
      includes the optional parameter ``<species_name>.beta`` where beta is equal to :math:`\frac{v}{c}`. The plasma
```

>       will be initialized to move at velocity beta*c in the

Reviewer: ```suggestion
      will be initialized to move at velocity :math:`\beta \cdot c` in the
```

      ``<species_name>.momentum_function_uy(x,y,z)`` and ``<species_name>.momentum_function_uz(x,y,z)``,
      which gives the distribution of each component of the momentum as a function of space.

* ``<species_name>.theta_distribution_type`` (`string`) optional (default ``constant``)
    Only read if ``<species_name>.momentum_distribution_type`` is ``maxwell_boltzmann`` or ``maxwell_juttner``.

>     * If ``constant``, use a constant temperature, given by the float ``<species_name>.theta``. The default value of ``<species_name>.theta`` is 10.

Reviewer: ```suggestion
    * If ``constant``, use a constant temperature, given by the float ``<species_name>.theta``. The default value of ``<species_name>.theta`` is ``10``.
```

Wouldn't `0` be a better default for `theta` though? `10` seems a bit arbitrary and could be surprising if the parameter is not set.

Reviewer: Discussed in today's dev meeting:
- let's add no default and require user input, this makes sure we don't have problems if users do typos in input ketys and/or forget to set them at all
- let's double-check the normalization of `theta` is well documented
- let's throw an error if the user input value is too small (e.g., <=1)
- let's throw a warning if the user input is clearly non-relativistic

Author: * Done. No default value, though constant distribution remains a default.
* It's discussed a few lines up at the top of the block.
* Also done (error for theta < 0)
* Error will throw for theta < 0.1 in MJ, and warning for theta >0.01 in MB, since that's where the differences between MJ and MB exceed ~1%

Author: I kept errors, but now the code will only warn for theta >0.01 during input parsing (and therefore only for the constant temperature case). Otherwise we'd have to call a warning from the GPU, and we'd also likely flood the user with warnings.
